src/MainWindow.vala

```diff
@@ -787,9 +788,11 @@ namespace PantheonTerminal {
                 }
             }
 
-            PantheonTerminal.saved_state.tabs = {};
+            PantheonTerminal.TerminalApp.saved_state.set_strv ("tabs", {});
+
+            int focus = PantheonTerminal.TerminalApp.saved_state.get_int ("focused-tab");
+            focus.clamp (0, tabs.length - 1);
```

Comment:
`clamp` returns you a new int here, so currently, you are clamping and then immediately discarding the result.
```suggestion
            focus = focus.clamp (0, tabs.length - 1);
```